jac-uk/admin
# Feature/1327 enable request of character check form completion

## What's included?
The Admin user is able to request candidates to complete a consent form from the tab Character checks on the platform.

I've also made small changes to the Agency report tab in Admin->Applications as it was not accommodating character information v2 financial issues and added Character checks tab to show information collected via consent form. Some of this information is duplicated in other tabs, so changes could be made if required to only show information that is not shown anywhere else.

**1. Before deployment - we need to set up Notify templates**

**2. UPDATE - @KateMJAC  @joy-ade  'Enable Character Checks' button was added to make sure functionality works for all exercises:**

![EnableCharChecks](https://user-images.githubusercontent.com/40855898/127454429-5d18282d-9760-42d0-bcc7-5a2b04f1e742.png)

## Who should test?
✅ Product owner
✅ Developers
✅ UTG

## How to test?
The functionality works with all stages except of 'Review' and with new applications only.

- Create a new exercise
- Apply for a vacancy
- Progress candidates from Review stage forward to any other stage
- Go to Tasks->Character checks. Candidates will display in Not requested tab.
- Select required candidate(s) and click Send requests.
- Read the pop-up message and click 'I confirm' button. There will be a status message displayed to advise if the operation was successful. 
- Click on Notifications tab on top of the page. Ready to go out emails will be displayed here and will be sent **in appx. 5 mins**.
- The candidates will be moved to 'Requested' tab on Character checks page and the date when request was made will be displayed. 
- Select required candidate(s) and click 'Send reminders' - the process is the same as above. The date when reminder was sent will be displayed.
- Check your emails (at the moment request/reminder emails are the same - _we will need to set up two templates on Notify_)
- Log in to Apply account and complete consent form
- When consent form is completed, the candidate will move to Character checks->'Completed' tab.

## Risk - how likely is this to impact other areas?
🔴 High risk - this includes a lot of changes to shared code

## Additional context
(Please test together with #1312 and [#773](https://github.com/jac-uk/apply/pull/791)) [Here is the video description](https://drive.google.com/file/d/1BGKzj_PvCIjhXHCVM4M6KM3DlR29_zvI/view?usp=sharing) of expected functionality

![Admin1](https://user-images.githubusercontent.com/40855898/125980993-eb092f1f-11bd-4e83-8848-8507031e2e86.png)
![Admin2](https://user-images.githubusercontent.com/40855898/125981000-b4f8e5d7-72a9-40d4-9124-5613d31afedb.png)
![Admin3](https://user-images.githubusercontent.com/40855898/125981010-732226b3-e821-4764-bdc4-9a61204e8002.png)
![Admin4](https://user-images.githubusercontent.com/40855898/125981016-8dfeb829-4b2a-4ef2-ba5b-b33b01fe07ec.png)
![Admin5](https://user-images.githubusercontent.com/40855898/125981028-843cb19d-6a0e-44a9-a626-01ad1eb14e84.png)

---
PREVIEW:DEVELOP
_can be OFF, DEVELOP or STAGING_
